### PR TITLE
HPCC-16979 Assert failure when can't load plugin for folding on OSX

### PR DESCRIPTION
**Pull request by richardkchapman**
A bug in the OSX compiler means we can't safely throw exceptions from
inside doFoldExternalCall. Refatory to ensure that any likely exceptions
are thrown before we enter that function.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday Please review

File: ecl/hql/hqlfold.cpp [hunk 1]
```diff
@@ -680,11 +680,10 @@ bool checkExternFoldable(IHqlExpression* expr, unsigned foldOptions, StringBuffe
     return true;
 }
 
-IValue * doFoldExternalCall(IHqlExpression* expr, unsigned foldOptions, ITemplateContext *templateContext, const char *library, const char *entrypoint)
+void *loadExternalEntryPoint(IHqlExpression* expr, unsigned foldOptions, ITemplateContext *templateContext, const char *library, const char *entrypoint, HINSTANCE &hDll)
```
Comment (ghalliday) [inline ecl/hql/hqlfold.cpp]: hDll parameter is not used - it still assigns to a local variable in the body of the function (line 714).

**Review by ghalliday — DISMISSED** [review]
@richardkchapman 
I'm not sure of hDLL should be unloaded in the exception case in the folder.  Might be worth double checking the logic at the same time.

**Comment by richardkchapman** [comment]
@ghalliday Please re-review

**Pull request merged**